lib/broccoli/glimmer-app.ts

```diff
@@ -206,7 +216,64 @@ export default class GlimmerApp extends AbstractBuild {
     };
   }
 
-  private cssTree(): Tree | null {
+  //
+  // TEMPLATE COMPILATION
+  //
+
+  /**
+   * True if the app should emit templates as Glimmer binary bytecode.
+   */
+  protected get emitBytecodeTemplates() {
+    return this.options.templateFormat === 'bytecode';
+  }
+
+  /**
+   * Given a Broccoli tree containing `.hbs` files, returns a tree with those
+   * templates transformed into their precompiled JSON form and saved as `.js`
+   * files.
+   */
+  protected compileTemplatesToJSON(appTree: Tree): Tree {
+    let glimmerEnv = this.getGlimmerEnvironment();
+    return new GlimmerJSONCompiler(appTree, {
+      rootName: this.project.pkg.name,
+      GlimmerENV: glimmerEnv
+    });
+  }
+
+  protected compileTemplatesToBytecode(appTree: Tree) {
+    let { project: { root } } = this;
+
+    // Template compiler needs access to root package.json
+    let pkgJsonTree = new UnwatchedDir(root);
+    pkgJsonTree = new Funnel(pkgJsonTree, {
+      include: ['package.json']
+    });
+
+    let templateTree: Tree = new MergeTrees([appTree, pkgJsonTree]);
+
+    let compiledOutput = new GlimmerBundleCompiler(templateTree, {
+      mode: 'module-unification',
+      outputFiles: {
+        heapFile: 'templates.gbx',
+        dataSegment: 'src/data-segment.js'
+      }
+    });
+
+    templateTree = new Funnel(compiledOutput, {
```

Comment:
Why split these and then just merge them back together on https://github.com/glimmerjs/glimmer-application-pipeline/pull/138/files#diff-7be832ed388b7459dc35fe197115744eR134

Reply:
@chadhietala The data segment tree is merged back into the other JavaScript so the module is available to Rollup, but the `.gbx` stays separate. I was trying to only pass `.js` files to Rollup so it's clear what the contract is.